### ISSUE TITLE
[a11y] Modal focus management

### ISSUE THREAD

**Issue by Cheffheid** [WebDevStudios/wd_s#227]
When using a button to toggle the modal with the included modal.js script the focus is never shifted to the modal itself. This means that tabbing will simply continue from where the button was.

Here's a video to illustrate the issue:  https://www.youtube.com/watch?v=fzXYWa_1Cf4

At the very least the keyboard focus needs to also be moved to the modal after the modal is toggled. As an additional note, keyboard focus should probably be restricted to the modal as well. Meaning that after "leaving" the last element the focus needs to return to the first element inside of the modal.

And finally, when closing the modal the focus needs to return back to the element that opened it in the first place.

**Comment by samikeijonen** [comment]
Everything @Cheffheid said.

Here is good example of [accessible modal dialog](http://edenspiekermann.github.io/a11y-dialog/).

**Comment by Cheffheid** [comment]
Thanks for that, @samikeijonen, very useful. :)

I've started a branch hotfix/modal-focus-fix for this, where I've pushed a number of changes to start fixing this.

The state of that branch is like this now:
https://youtu.be/QOcjDux9w3M

The theme itself doesn't include any markup for the modals themselves, so I'm not sure if it's worthwhile to do things like toggling the aria-hidden attribute (and it needs to be done if there is one, because in NVDA with Chrome that causes issues: https://www.youtube.com/watch?v=GVn4KwFUdT0). 

If need be, I can provide a hard coded PHP template that I've used to test with.
